spec/integration/user_post_spec.rb

```diff
@@ -1,5 +1,5 @@
 require 'rails_helper'
-
+# integration tests for posts a users
 RSpec.describe 'User Integration', type: :system do
   before do
     @user1 = User.create(id: 100, name: 'Username 1', photo: 'test1.jpg', postcounter: 10, bio: 'User 1 bio')
```

Comment:
While you did a good job working on the project requirements you missed out some tests there are a total number of 32 features that you are supposed to test but for you you only have 8 examples that are passing. And also you should have separate files for the user tests and post tests but for you I can see you used only one file to write tests for both.

Please use these guidelines to write the tests:
2. Has featured tests for User index page:

Spec file should only test User index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect
Tests that asserts you can see the username of all other users.
Tests that asserts you can see the profile picture for each user.
Tests that asserts you can see the number of posts each user has written.
Tests that redirects to the user's show page when you click on a user.


3. Has featured tests for User show page:

Spec file should only test User show page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for each redirect
Tests that asserts you can see the user's profile picture.
Tests that asserts you can see the user's username.
Tests that asserts you can see the number of posts the user has written.
Tests that asserts you can see the user's bio.
Tests that asserts you can see the user's first 3 posts.
Tests that asserts you can see a button that lets me view all of a user's posts.
Tests that redirects to the post's show page when you click on a user's post.
Tests that redirects to the user's post's index page when you click on a see all posts.


4. Has featured tests for User post index page:

Spec file should only test User post index page. It can be only one spec testing all off the things below or more, ideally one for the asserts and one for the redirect
Tests that asserts you can see the user's profile picture.
Tests that asserts you can see the user's username.
Tests that asserts you can see the number of posts the user has written.
Tests that asserts you can see a post's title.
Tests that asserts you can see some of the post's body.
Tests that asserts you can see the first comments on a post.
Tests that asserts you can see how many comments a post has.
Tests that asserts you can see how many likes a post has.
Tests that asserts you can see a section for pagination if there are more posts than fit on the view.
Tests that redirects to the post's show page when you click on a post.


5. Has featured tests for User post show page:

Spec file should only test User show page. It can be only one spec testing all off the things below or more.
Tests that asserts you can see the post's title.
Tests that asserts you can see who wrote the post.
Tests that asserts you can see how many comments it has.
Tests that asserts you can see how many likes it has.
Tests that asserts you can see the post body.
Tests that asserts you can see the username of each commentor.
Tests that asserts you can see the comment each commentor left.


Hint:  If the student is struggling with tests, here's a [capybara cheat sheet](https://gist.github.com/zhengjia/428105) that can share to help unblock them.